### ISSUE TITLE
Errors that occur when deleting release

### ISSUE THREAD

**Issue by ghost** [go-gitea/gitea#4205]
- Gitea version (or commit ref): f54626d (After #3981)
- Git version: 2.16.3
- Operating system: macOS
- Database:
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (I think yes, because the page is not refreshed)
  - [ ] No
  - [ ] Not relevant
- Log gist:
```
[Macaron] PANIC: runtime error: invalid memory address or nil pointer dereference
/usr/local/Cellar/go/1.10.2/libexec/src/runtime/panic.go:502 (0x402d248)
	gopanic: reflectcall(nil, unsafe.Pointer(d.fn), deferArgs(d), uint32(d.siz), uint32(d.siz))
/usr/local/Cellar/go/1.10.2/libexec/src/runtime/panic.go:63 (0x402c2bd)
	panicmem: panic(memoryError)
/usr/local/Cellar/go/1.10.2/libexec/src/runtime/signal_unix.go:388 (0x4043319)
	sigpanic: panicmem()
/Users/yasuokav/go/src/code.gitea.io/gitea/models/access.go:64 (0x4913ed6)
	accessLevel: if !repo.IsPrivate {
/Users/yasuokav/go/src/code.gitea.io/gitea/models/release.go:451 (0x4981de3)
	DeleteReleaseByID: mode, _ := accessLevel(x, u.ID, rel.Repo)
/Users/yasuokav/go/src/code.gitea.io/gitea/routers/repo/release.go:288 (0x4ad52fd)
	DeleteRelease: if err := models.DeleteReleaseByID(ctx.QueryInt64("id"), ctx.User, true); err != nil {
```
## Description
As title. The release will be deleted, but the attachments did not delete(this is another issue #2618).

## Screenshots
N/A

**Comment by techknowlogick** [comment]
Closed with #4207
